doc/source/walkthrough.rst

```diff
@@ -146,64 +283,116 @@ resources.
+.. tabs::
+  .. group-tab:: Python
+
+    .. code-block:: python
+
+      # Ray also supports fractional resource requirements
+      @ray.remote(num_gpus=0.5)
+      def h():
+          return 1
+
+      # Ray support custom resources too.
+      @ray.remote(resources={'Custom': 1})
+      def f():
+          return 1
+
+  .. group-tab:: Java
+
+    .. code-block:: java
+
+      // Ray aslo supports fractional and custom resources.
+      Map<String, Double> resources = ImmutableMap.of("GPU", 0.5, "Custom", 1);
+      CallOptions options = new CallOptions.Builder()
+        .setResources(resources).createCallOptions();
+      Ray.call(MyRayApp::foo, callOptions);
+
+Multiple returns
+~~~~~~~~~~~~~~~~
+
+.. tabs::
+  .. group-tab:: Python
+
+    Python remote functions can return multiple object IDs.
+
+    .. code-block:: python
 
-  # Ray also supports fractional resource requirements
-  @ray.remote(num_gpus=0.5)
-  def h():
-      return 1
+      @ray.remote(num_return_vals=3)
+      def return_multiple():
+          return 1, 2, 3
 
-  # Ray support custom resources too.
-  @ray.remote(resources={'Custom': 1})
-  def f():
-      return 1
+      a_id, b_id, c_id = return_multiple.remote()
 
-Further, remote functions can return multiple object IDs.
+  .. group-tab:: Java
 
-.. code-block:: python
+    Java remote functions doesn't support returning multiple objects.
 
-  @ray.remote(num_return_vals=3)
-  def return_multiple():
-      return 1, 2, 3
+Cancelling tasks
+~~~~~~~~~~~~~~~~
 
-  a_id, b_id, c_id = return_multiple.remote()
+.. tabs::
+  .. group-tab:: Python
 
-Remote functions can be canceled by calling ``ray.cancel`` on the returned Object ID. Remote actor functions can be stopped by killing the actor using the ``ray.kill`` interface.
+    Remote functions can be canceled by calling ``ray.cancel`` on the returned Object ID. Remote actor functions can be stopped by killing the actor using the ``ray.kill`` interface.
 
-.. code-block:: python
+    .. code-block:: python
 
-  @ray.remote
-  def blocking_operation():
-      time.sleep(10e6)
-      return 100
-  
-  obj_id = blocking_operation.remote()
-  ray.cancel(obj_id)
+      @ray.remote
+      def blocking_operation():
+          time.sleep(10e6)
+          return 100
 
-.. autofunction:: ray.cancel
-    :noindex:
+      obj_id = blocking_operation.remote()
+      ray.cancel(obj_id)
+
+    .. autofunction:: ray.cancel
+        :noindex:
+
+  .. group-tab:: Java
+
+    Cancelling tasks hasn't been implemented in Java yet.
 
 Objects in Ray
 --------------
 
-In Ray, we can create and compute on objects. We refer to these objects as **remote objects**, and we use **object IDs** to refer to them. Remote objects are stored in `shared-memory <https://en.wikipedia.org/wiki/Shared_memory>`__ **object stores**, and there is one object store per node in the cluster. In the cluster setting, we may not actually know which machine each object lives on.
+.. tabs::
+
+  .. group-tab:: Python
```

Comment:
Since the semantics are the same here, it's just minor changes to the API calls, can we just add a code block for the call as opposed to changing the whole paragraph?

Having to maintain huge groups of text between the two is going to be very difficult.

For instance, here it's probably easier to say 

>We refer to these objects as **remote objects**, and we use **object IDs** to refer to them. Remote objects are stored ...

as 
> We refer to these objects as **remote objects**, and we use **object IDs** (or ``RayObject`` in Java) to refer to them. Remote objects are stored....